src/installation/guides/chroot.md

```diff
@@ -0,0 +1,222 @@
+# Installation via chroot (x86 or x86_64)
+
+## Prepare Filesystems
+
+These instructions can be followed on any running Linux system that has xbps
+installed, including an already working installation of Void or an official
+[live image](../live-images/prep.md). It is assumed that the host computer is
+connected to the Internet.
+
+[Partition your disks](../live-images/partitions.md) and format them using
+[mke2fs(8)](https://man.voidlinux.org/mke2fs.8),
+[mkfs.xfs(8)](https://man.voidlinux.org/mkfs.xfs.8),
+[mkfs.btrfs(8)](https://man.voidlinux.org/mkfs.btrfs.8) or whatever tools are
+necessary for your filesystem(s) of choice.
+
+[cfdisk(8)](https://man.voidlinux.org/cfdisk.8) and
+[fdisk(8)](https://man.voidlinux.org/fdisk.8) are available on live
+distributions for partitioning, but you may wish to use
+[gdisk(8)](https://man.voidlinux.org/gdisk.8) (from the package `gptfdisk`) or
+[parted(8)](https://man.voidlinux.org/parted.8) instead.
+
+> Note: For a UEFI booting system, make sure to create a UEFI System Partition
+> (ESP). The ESP should have the partition type "EFI System" (code `ef00`) and
+> be formatted as fat32 using
+> [mkfs.vfat(8)](https://man.voidlinux.org/mkfs.vfat.8).
+
+### Create a New Root and Mount Filesystems
+
+This guide will assume the new root filesystem is mounted on `/mnt`. You may
+wish to mount it elsewhere.
+
+If using UEFI, mount the EFI System Partition as `/mnt/boot/efi`.
+
+For example, if `/dev/sda2` is to be mounted as `/` and `/dev/sda3` is to be
+booted as `/home` and `dev/sda1` is the EFI System Partition
+
+```
+# mount /dev/sda2 /mnt/
+# mkdir /mnt/boot/
+# mount /dev/sda3 /mnt/home/
+# mkdir -p /mnt/boot/efi/
+# mount /dev/sda1 /mnt/boot/efi/
+```
+
+Initialize swap space, if desired using
+[mkswap(8)](https://man.voidlinux.org/mkswap.8).
+
+## Base Installation
+
+[Select a mirror to download packages
+from.](../../xbps/repositories/mirror/index.md) For simplicity, save this URL to
+a shell variable for later use. Append `/current` to this repository url to
+install a glibc distribution, append `/current/musl` for a musl distribution.
+
+> Note: musl binaries are not available for i686.
+
+For example, this is the url for glibc from the Tier 1 server in Finland,
+
+```
+# REPO=https://alpha.de.repo.voidlinux.org/current
+```
+
+xbps also needs to know what architecture is being installed, available options
+are `x86_64`, `x86_64-musl` and `i686` for PC architecture computers.
+
+For example,
+
+```
+# ARCH=x86_64
+```
+
+This architecture must be compatible with your current operating system, but
+does not need to be the same. If your host is running an x86_64 operating
+system, any of the three architectures can be installed (whether the host is
+musl or glibc), but an i686 host can only install i686 distributions.
+
+Use xbps to boot strap the distribution by installing the `base-system`
+metapackage.
+
+```
+# XBPS_ARCH=$ARCH xbps-install -S -r /mnt -R "$REPO" base-system
+```
```

Comment:
Given that there are ROOTFS tarballs available in the downloads section, we could tell people that they can also use that instead of xbps. They'd get outdate packages, but it is a complete system as well. Could be listed as the alternate install method: you either need a working version of xbps and an internet connection or the ROOTFS tarball and working versions or tar and xz.

Unpacking it requires only

```
# cd /mnt
# tar xvf void-x86_64-20019...-ROOTFS.tar.xz
```

Reply:
Or `tar xvf rootfs.tar.xz -C /mnt` for a more correct solution.

Reply:
Okay, I spend a lot of time thinking about whether this belongs in the same guide or its own and I'm leaning toward agreeing with you. I haven't really used rootfs tarballs, but I can see how it would appeal to someone who doesn't already have a working void distribution.

The basic process would be something like this, right?
- create partitions and filesystems
- unpack the tarball
- chroot
- distribution configuration
- xbps-install -S base-system

Reply:
Up to the part where you're mounting the EFI and home partitions, the guide is exactly the same. Then you:

* unpack the tarball using the command I listed above (it's important to unpack it as root, otherwise the files will be all wrong)
* chroot the same way you had explained (copying resolv.conf and mounting the filesystems)
* install the kernel you want, because the rootfs is nearly a base-system already, just without a kernel

And then it's back to your guide.

Reply:
I'm going to get to this today, if I can. If hopefully next saturday at the latest.

Reply:
No problem!

Reply:
Might also be nice to mention that if you use the ROOTFS tarball you don't need to install from a void system/live image.  I had a grml live usb lying around so I just used that.

Reply:
just an update--- I'm actually getting to this today. @ifreund--- yeah, the introductory paragraph is going to be rewritten a bit so that this is clear.